pkg/api/norman/customization/gke/listers.go

```diff
@@ -228,3 +229,41 @@ func listSharedNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, er
 
 	return data, http.StatusOK, err
 }
+
+func listClusters(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	if cap.ProjectID == "" {
+		return nil, httperror.ServerError.Status, fmt.Errorf("projectId is required")
+	}
+
+	if cap.Region == "" && cap.Zone == "" {
+		return nil, httperror.ServerError.Status, fmt.Errorf("either region or zone is required")
+	}
+
+	var location string
+	if cap.Region != "" && cap.Zone != "" {
+		logrus.Debugf("found both region and zone in /meta/gkeClusters request, using region")
+	}
+	if cap.Region != "" {
+		location = cap.Region
+	} else {
+		location = cap.Zone
+	}
+
+	client, err := getContainerServiceClient(ctx, cap.Credentials)
+	if err != nil {
+		return nil, httperror.ServerError.Status, err
+	}
+
+	parent := "projects/" + cap.ProjectID + "/locations/" + location
```

Comment:
So if region is present this list gets more specific ? Does Wes know that they can pass in either/both ?

Reply:
It's not "more specific", it's either/or. A cluster is either regional or zonal. It would not make sense to pass both in. However, I'm proposing that in case both are passed in, we default to looking up clusters for the region and ignoring the zone. The other option is to return an error if both region and zone are provided. @westlywright what do you think?

Reply:
Lets just do region in the UI then, thats what eks does. We can leave API accepting either, and prefer region.

Reply:
Sorry, no, I wasn't clear - I don't mean "either/or" as in it doesn't matter, I mean "either/or" as in it's either one or the other, not both. It would make no sense to GKE to only be able to select region in the UI. You need to be able to select one or the other. A cluster in region "us-west1" is not the same as a cluster in the zone "us-west1-a".

![image](https://user-images.githubusercontent.com/1220710/113319742-9f604b00-92c6-11eb-8598-1ab96e2f92c7.png)